nestor_api/lib/k8s/builders.py

```diff
@@ -130,3 +131,67 @@ def get_variables(app_config: dict) -> dict:
         **app_variables,
         **ope_variables,
     }
+
+
+def set_namespace(app_config: dict, resources: list, templates: dict) -> Optional[dict]:
+    """Attach the expected namespace to the deployment and return the namespace"""
+    namespace = None
+    namespace_name = app_config.get("namespace")
+
+    if namespace_name is not None:
+        for resource in resources:
+            resource["metadata"]["namespace"] = namespace_name
+        namespace = yaml_lib.parse_yaml(templates["namespace"]({"name": namespace_name}),)
+
+    return namespace
+
+
+# pylint: disable=bad-continuation
+def set_replicas(
+    app_config: dict, process_name: str, recipe: dict, templates: dict
+) -> Optional[dict]:
+    """Attach the correct Replicas values to the recipe object and return the corresponding
+    `hpa` if applicable"""
+    scales = app_config["scales"]
+    scales_config = scales[process_name] if process_name in scales else scales["default"]
```

Comment:
Maybe ?
```suggestion
    scales_config = scales[process_name] if process_name in scales else scales.get("default", {})
```
so that it works with the default values if the `default` field is not defined

Reply:
We want this to throw because it is mandatory, I can do a custom error if you think it is better